src/backend/executor/bridge.ts

```diff
@@ -0,0 +1,474 @@
+
+        const process = new ScheduledProcess(commandLine, { processType: ProcessType.parse });
+
+        // TODO: Find a better way to collect full logger output
+        let processOutput = '';
+
+        process.processStdout((output) => processOutput += output);
+
+        process.processStatusChange((status) => {
+            if (status === ProcessStatus.finished) {
+                ExtensionApi.diagnostics.parseDiagnosticsData(processOutput);
+            }
+        });
+
+        ExtensionApi.executorManager.addToQueue(process, 'replace');
+    }
+
+    public async checkVersion(): Promise<boolean> {
+        return new Promise((res, _rej) => {
+            if (this.versionChecked) {
+                res(this.versionChecked);
+                return;
+            }
+
+            const commandLine = this.getVersionCmdLine();
+
+            if (commandLine === undefined) {
+                this._bridgeMessages.fire('>>> Unable to determine CodeChecker version commandline\n');
+
+                this.versionChecked = false;
+                return;
+            }
+
+            const process = new ScheduledProcess(commandLine, { processType: ProcessType.version });
+
+            let processOutput = '';
+
+            process.processStdout((output) => processOutput += output);
+
+            process.processStatusChange(async (status) => {
+                switch (status) {
+                case ProcessStatus.running: return;
+                case ProcessStatus.finished:
+                    try {
+                        // Structure: CodeChecker analyzer version: \n {"Base package version": "M.m.p", ...}
+                        // eslint-disable-next-line @typescript-eslint/naming-convention
+                        const data = JSON.parse(processOutput) as { 'Base package version': string };
+                        // Convert semver to array
+                        const version = data['Base package version'].split('.').map(x => parseInt(x));
+
+                        const minimum = [6, 18, 1];
+
+                        if (version < minimum) {
+                            this._bridgeMessages.fire(`>>> Unsupported CodeChecker version ${version}\n`);
+                            this._bridgeMessages.fire(`>>> Minimum version: ${minimum}\n`);
+
+                            this.versionChecked = false;
+
+                            if (!this.shownVersionWarning) {
+                                this.shownVersionWarning = true;
+                                let choice;
+
+                                while (choice !== 'Close') {
+                                    choice = await window.showWarningMessage(
+                                        `CodeChecker: Version ${minimum.join('.')} not supported. ` +
+                                            'Update CodeChecker, or check the extension settings.',
+                                        'Open releases',
+                                        'Installation guide',
```

Comment:
The message is not correct, because you say that the minimum supported version is not supported. I recommend to refactor this message to something like this:
```suggestion
                                        `CodeChecker version you are using (${version.join('.')}) is not supported. ` +
                                        `The minimum supported version is ${minimum.join('.')}. Please update to the ` + 
                                        `latest CodeChecker version, or check the extension settings.`,
                                        'Open releases',
                                        'Installation guide',
                                        'Settings',
```

And also add an extra button to open the Settings of this extension.

Reply:
Fixed as requested, and added the Open settings button to the "Not found" notif as well.